v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -214,6 +214,22 @@ describe("main", function () {
         flag: true,
       });
     });
+
+    it("should not fail when a global parameter is not recognized (it might be a task parameter)", async function () {
+      const command = "npx hardhat task --nonExistingFlag <value>";
```

Comment:
Is this really a task level flag, rather than a non-existant flag? So `--taskLevelFlag` or something?

Reply:
[Done in this commit](https://github.com/NomicFoundation/hardhat/pull/5373/commits/d299a671705ff843a5b1717e09b0760f1abde339)